src/components/VariantDb/index.tsx

```diff
@@ -0,0 +1,245 @@
+import * as React from 'react';
+import { Button, Col, List, notification, Row, Typography } from 'antd';
+import { deleteCluster, getStatus, launchCluster } from './fetchVariantCluster';
+import { MAX_MINUTES_TRY, INCREMENT, clusterStatus, isInterimState } from './store';
+
+import './index.css';
+import LaunchClusterCard from './LaunchClusterCard';
+
+const { Title } = Typography;
+
+type VariantDbProps = {
+  api: Function;
+};
+
+type VariantDbState = {
+  status: string;
+  modalVisible: boolean;
+};
+
+class VariantDb extends React.Component<VariantDbProps, VariantDbState> {
+  state = {
+    status: '',
+    modalVisible: false,
+  };
+
+  data = [
+    {
+      name: 'Studies',
+      value: 13,
+    },
+    {
+      name: 'Participants',
+      value: <div>14494</div>,
+    },
+    {
+      name: 'Genes',
+      value: 21393,
+    },
+    {
+      name: 'Variants',
+      value: 29848393,
+    },
+    {
+      name: 'Exomic variants',
+      value: 2387298,
+    },
+  ];
+
+  errorNotification = (message: string, decription: string) => {
+    notification.error({
+      message: message,
+      description: decription,
+    });
+  };
+
+  async componentDidMount() {
+    const { api } = this.props;
+    try {
+      const res = await getStatus(api);
+      if (isInterimState(res.status)) {
+        this.resolveInterimState(api);
+      } else if (res.status === clusterStatus.rollback) {
+        await deleteCluster(api);
+        this.resolveInterimState(api);
+      }
+
+      this.setState({
+        status: res.status,
+      });
+    } catch (e) {
+      this.errorNotification(`Error ${e.response?.status || ''}`, 'Cannot connect with cluster');
+    }
+  }
+
+  resolveInterimState = (api: Function) => {
+    let interval: NodeJS.Timeout;
+    let counter = 1;
+
+    const verifyStatus = async () => {
+      if (counter * INCREMENT > MAX_MINUTES_TRY * 60 * 1000) {
```

Comment:
I would put  MAX_MINUTES_TRY * 60 * 1000 in a variable with a meaningful name where the time unit is explicit: MAX_TRIES_DURATION_IN_MS 

I think parenthesis would add readability (counter * Increment) > MAX_TRIES_DURATION_IN_MS

Reply:
In case of error. I you sure that clearInterval is always called ? (avoid possible memory leak)

Reply:
Prettify will not let me do that for the parenthesis.